src/scrolltext.cpp

```diff
@@ -0,0 +1,134 @@
+#include "scrolltext.h"
+
+
+ScrollText::ScrollText(QWidget *parent) :
+    QWidget(parent), scrollPos(0)
+{
+    staticText.setTextFormat(Qt::PlainText);
+
+//    setFixedHeight(fontMetrics().height()*2); //The theme sets this
+    leftMargin = height() / 3;
+
+    setSeparator("   ---   ");
+
+    connect(&timer, SIGNAL(timeout()), this, SLOT(timer_timeout()));
+    timer.setInterval(50);
+}
+
+QString ScrollText::text() const
+{
+    return _text;
+}
+
+void ScrollText::setText(QString text)
+{
+    _text = text;
+    updateText();
+    update();
+}
+
+QString ScrollText::separator() const
+{
+    return _separator;
+}
+
+void ScrollText::setSeparator(QString separator)
+{
+    _separator = separator;
+    updateText();
+    update();
+}
+
+void ScrollText::updateText()
+{
+    timer.stop();
+
+    singleTextWidth = fontMetrics().horizontalAdvance(_text);
+    scrollEnabled = (singleTextWidth > width() - leftMargin*2);
+
+    if(scrollEnabled)
+    {
+        scrollPos = -64;
```

Comment:
I actually have smooth scrolling code from the AC prototype if you're ever interested.

Reply:
That could work if this scroll text doesn't work out for us